LLMonFHIR/FHIR Display/OpenAIChatView.swift

```diff
@@ -87,4 +77,112 @@ struct OpenAIChatView: View {
+            Chat(role: .user, content: chat.last?.content ?? ""),
+            Chat(role: .system, content: String(localized: "FUNCTION_CONTEXT") + stringResourcesArray.rawValue)
+        ]
+        
+        let chatStreamResults = try await openAPIComponent.queryAPI(withChat: chat, withFunction: functions)
+        
+        class ChatFunctionCall {
+            var name: String = "", arguments: String = "", finishReason: String = ""
+        }
+        
+        var functionCall = ChatFunctionCall()
+        
+        for try await chatStreamResult in chatStreamResults {
+            for choice in chatStreamResult.choices {
+                if let deltaName = choice.delta.name {
+                    functionCall.name += deltaName
+                }
+                if let deltaArguments = choice.delta.functionCall?.arguments {
+                    functionCall.arguments += deltaArguments
+                }
+                if let finishReason = choice.finishReason {
+                    functionCall.finishReason += finishReason
+                    if finishReason == "get_resource_titles" { break }
+                }
+            }
+        }
+        
+        var functionCallOutputArray = [String]()
+        
+        if functionCall.finishReason == "function_call" {
+            let trimmedArguments = functionCall.arguments.trimmingCharacters(in: .whitespacesAndNewlines)
+            
+            if let resourcesRange = trimmedArguments.range(of: "\"resources\": \"([^\"]+)\"", options: .regularExpression) {
+                functionCallOutputArray = trimmedArguments[resourcesRange]
+                    .replacingOccurrences(of: "\"resources\": \"", with: "")
+                    .replacingOccurrences(of: "\"", with: "")
+                    .components(separatedBy: ",")
+            }
+        }
+        
+        return functionCallOutputArray
+    }
+
+    private func processFunctionCallOutputArray(functionCallOutputArray: [String], resourcesArray: [FHIRResource]) {
+        let stringResourcesArray = resourcesArray.map { "\($0.displayName) in \($0.resourceType)" }
+        for resource in functionCallOutputArray {
+            var stringResource = resource
+            stringResource = resource.trimmingCharacters(in: .whitespaces)
+            if let index = stringResourcesArray.firstIndex(of: stringResource) {
+                let resourceJSON = resourcesArray[index].jsonDescription
+                let functionContent = """
+                Based on the function get_resource_titles you have requested the following health records: \(stringResource).
+                This is the associated JSON data for the resources which you will use to answer the users question: \(resourceJSON).
+                Use this health record to answer the users question ONLY IF the health record is applicable to the question.
+                """
+                chat.append(Chat(role: .function, content: functionContent, name: "get_resource_titles"))
```

Comment:
I think that the response should rather be a `.system` message if I correctly understood this from the documentation. 
Appending multiple responses sounds good to me 👍